src/foam/mlang/mlang.js

```diff
@@ -317,7 +317,8 @@ foam.CLASS({
       return foam.String.constantize(this.cls_.name) + '(' +
           this.arg1.toString() + ', ' +
           this.arg2.toString() + ')';
-    }
+    },
+    function contradicts(other) { return false; }
```

Comment:
With an AND specific a.reduceWithAnd(b), just avoid returning AND.
This method is basically giving mlangs a way to express their AND partialEval rules (FOAM1 had these embedded inside AND). "When combined with other mlang b, what happens?"